src/JsonRpcEngine.ts

```diff
@@ -593,20 +593,20 @@ export class JsonRpcEngine extends SafeEventEmitter {
     isComplete: boolean,
   ): void {
     if (!hasProperty(res, 'result') && !hasProperty(res, 'error')) {
-      throw new EthereumRpcError(
+      throw new JsonRpcError(
         errorCodes.rpc.internal,
         `JsonRpcEngine: Response has no error or result for request:\n${jsonify(
           req,
         )}`,
-        { request: req },
+        { request: req as Json },
```

Comment:
Not really related to your change, but I would prefer using `req` and `res` everywhere. Must've missed these when working on the module template update.